File: SeongYeon/ts/leetcode/20_ValidParentheses.ts
function isValid(s: string): boolean {
  const leftBracket = ['(', '{', '['];
  const rightBracket = [')', '}', ']'];
  const stack = [];

  for (const bracket of s) {
    if (leftBracket.includes(bracket)) stack.push(bracket);
    else {
      const idx = rightBracket.indexOf(bracket);
      if (stack.pop() !== leftBracket[idx]) return false;
    }
  }

  return !stack.length;
}

File: jiwon/leetcode/20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        for i in s:
            if i in '({[':
                stack.append(i)
            else:
                if not stack:
                    return False
                elif stack[-1] == '(' and i == ')':
                    stack.pop()
                elif stack[-1] == '{' and i == '}':
                    stack.pop()
                elif stack[-1] == '[' and i == ']':
                    stack.pop()
                else:
                    return False
        return False if stack else True
